Add impls for some easy bits

jeff-hiner opened the pull request:
Impls for capacity, clear, drain, is_empty, remove_entry, remove_entry_alt, reserve, shrink_to_fit were pretty straightforward. Implementations of something like iter_mut are going to require some thought and some new types, because naively allowing interior mutability of K2 within value_map will break invariants.

Partially addresses https://github.com/rust-embedded-community/multi-map/issues/13

Commit:
Add impls for capacity, clear, drain, is_empty, remove_entry, remove_entry_alt, reserve, shrink_to_fit

eldruin commented on src/lib.rs:
````diff
@@ -260,6 +260,114 @@ where
+    /// assert!(a.is_empty());
+    /// ```
+    pub fn drain(&mut self) -> hash_map::Drain<'_, K1, (K2, V)> {
+        self.key_map.clear();
````
Shouldn't this be `drain`?
```suggestion
        self.key_map.drain();
```

jeff-hiner replied:
No. Since one of the maps contains the set of the three values K1, K2, V we can simply return the drain of that one. Draining has the side effect of clearing the map, so for consistency we also must clear our adjacent `key_map`.

As an aside, this is another case where for ergonomics we should have a Drain struct that produces output to match `insert` (a remapped tuple from `(K1, (K2, V))` into  `(K1, K2, V)`), but we'd have to create it from scratch because `hash_map::Drain` isn't directly suitable for this.

eldruin replied:
Ah, I see. Thanks.

eldruin reviewed (commented):
Looks good to me, thanks! Could you also add tests (or more doctests at least) to cover all new functions?

jeff-hiner committed:
Add rustdoc

jeff-hiner committed:
Fix clippy lints in unit test

jeff-hiner commented:
Apologies this took so long. I've added doctests for the new functions, and also added a `len` function conceptually matching the one in HashMap (necessary for some of the doctests to make sense).

eldruin reviewed (approved):
Looks good to me, thank you!
